sklearn_numba_dpex/kmeans/drivers.py

```diff
@@ -799,11 +790,6 @@ def _set_dtype(self, X, sample_weight, centers_init):
                 f"will default back to float32 type."
             )
             compute_dtype = np.float32
-        elif compute_dtype != X.dtype:
-            text = (
-                f"KMeans is set to run with dtype {compute_dtype} but the data has "
-                f"been submitted with type {X.dtype}."
-            )
```

Comment:
I think we need to add a test that now access that `compute_dtype != X.dtype` is supported.

Reply:
Answering this comment and the following: the driver exposes a "dtype" option to force the `compute_dtype` no matter what the dtype of the input is. In practice it is used nowhere. We used in the benchmark in the very early commits of the repo.

What I propose here is to remove this option entirely and only set the driver to use for compute the `dtype` the data has been submitted. (except only for the case where the dtype is not supoported by the device)